### PR TITLE
Hotfix/sg 000 fix yolox replace head

### PR DESCRIPTION
**Pull request by BloodAxe**
Fixes bug in yolox's replace head method and added test that checks that
Removed removal of checkpoints cache dir since other tests may reuse the data and not re-download them (saving $$$ on S3 transfer)

### CODE REVIEW

File: tests/unit_tests/replace_head_test.py [hunk 1]
```diff
@@ -37,10 +46,6 @@ def test_dekr_replace_head(self):
         self.assertEqual(heatmap.size(1), 20 + 1)
         self.assertEqual(offsets.size(1), 20 * 2)
 
-    def tearDown(self) -> None:
-        if os.path.exists("~/.cache/torch/hub/"):
-            shutil.rmtree("~/.cache/torch/hub/")
-
```
Comment (Louis-Dupont) [inline tests/unit_tests/replace_head_test.py:43]: Any reason to remove this ?

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 2]
```diff
@@ -691,7 +696,7 @@ def replace_head(self, new_num_classes=None, new_head=None):
 
             new_last_layer = DetectX(
                 num_classes=new_num_classes,
-                stride=self._head.anchors.stride,
+                stride=self.strides,
```
Comment (Louis-Dupont) [inline src/super_gradients/training/models/detection_models/yolo_base.py:699]: Not sure if I understand, this single line was the bug right? It was just forgotten in the previous PR right?

File: src/super_gradients/training/models/detection_models/yolo_base.py [hunk 3]
```diff
@@ -205,7 +205,12 @@ def __init__(
         self.n_anchors = 1
         self.grid = [torch.zeros(1)] * self.detection_layers_num  # init grid
 
-        self.register_buffer("stride", torch.tensor(stride), persistent=False)
+        if torch.is_tensor(stride):
+            stride = stride.clone().detach()
+        else:
+            stride = torch.tensor(stride)
+
+        self.register_buffer("stride", stride, persistent=False)
```
Comment (Louis-Dupont) [inline src/super_gradients/training/models/detection_models/yolo_base.py:213]: Is this just optimization, or it also meant to fix the bug/another bug?
I think that the only noticeable difference is if stride is on GPU, `stride.clone().detach()` will create a copy on GPU, while `torch.tensor(stride)` will create a copy on CPU. Was that the motivation?

**Review by Louis-Dupont — APPROVED** [review]
LGTM, I just asked minor clarification

File: tests/unit_tests/replace_head_test.py (hunk 1 above)
Comment (BloodAxe) [reply tests/unit_tests/replace_head_test.py:43]: Yes, as stated in the PR description above: 
> Removed removal of checkpoints cache dir since other tests may reuse the data and not re-download them (saving $$$ on S3 transfer)

:)

File: src/super_gradients/training/models/detection_models/yolo_base.py (hunk 2 above)
Comment (BloodAxe) [reply src/super_gradients/training/models/detection_models/yolo_base.py:699]: Exactly. Removed the self._head.anchors.stride, but did forget to test replace_head still works

File: src/super_gradients/training/models/detection_models/yolo_base.py (hunk 3 above)
Comment (BloodAxe) [reply src/super_gradients/training/models/detection_models/yolo_base.py:213]: Not exactly - this is no avoid a warning message if input is already a torch tensor.
Otherwise torch complains and suggest we use `x.clone().detach()` instead of `torch.tensor(x)`

File: tests/unit_tests/replace_head_test.py (hunk 1 above)
Comment (Louis-Dupont) [reply tests/unit_tests/replace_head_test.py:43]: Haha indeed, sorry 😅
